dvc/lock.py

```diff
@@ -173,11 +184,16 @@ def __init__(self, lockfile, tmp_dir=None, **kwargs):
         self._separator = flufl.lock.SEP
         self._set_claimfile()
         self._owned = True
+        self._wait = wait
         self._retry_errnos = []
 
     def lock(self, timeout: Optional[Union[timedelta, int]] = None):
         try:
-            super().lock(timeout or timedelta(seconds=DEFAULT_TIMEOUT))
+            if timeout is None:
+                # Default timeout if not explicitly passed
+                timeout = timedelta(seconds=DEFAULT_TIMEOUT)
```

Comment:
On `self._wait=True`, this will still use the default, no?

Shouldn't this be following:

```python
if not self._wait:
  timeout = timeout or timedelta(seconds=DEFAULT_TIMEOUT)
```